### PR TITLE
setNumPbhPerMonth and associated tests

### PR DESCRIPTION
**Pull request by 0xForerunner**

### CODE REVIEW

**Review by 0xOsiris — APPROVED** [review]

File: contracts/src/helpers/PBHExternalNullifier.sol [hunk 1]
```diff
@@ -55,6 +55,6 @@ library PBHExternalNullifier {
         (uint8 pbhNonce, uint8 month, uint16 year) = PBHExternalNullifier.decode(externalNullifier);
         require(year == BokkyPooBahsDateTimeLibrary.getYear(block.timestamp), InvalidExternalNullifierYear());
         require(month == BokkyPooBahsDateTimeLibrary.getMonth(block.timestamp), InvalidExternalNullifierMonth());
-        require(pbhNonce <= numPbhPerMonth, InvalidPbhNonce());
+        require(pbhNonce < numPbhPerMonth, InvalidPbhNonce());
```
Comment (0xKitsune) [inline contracts/src/helpers/PBHExternalNullifier.sol:58]: I might be out of the loop, can you clarify why this shouldnt be `<=`?

File: contracts/test/PBHVerifierVerify.sol [hunk 2]
```diff
@@ -75,4 +69,37 @@ contract PBHVerifierVerify is PBHVerifierTest {
...
+    /// @notice Test that setNumPBHPerMonth works as expected
+    function testSetNumPBHPerMonth() public {
+        MockWorldIDGroups(address(worldID)).setVerifyProofSuccess(true);
+        uint8 month = uint8(BokkyPooBahsDateTimeLibrary.getMonth(block.timestamp));
+        uint16 year = uint16(BokkyPooBahsDateTimeLibrary.getYear(block.timestamp));
+
+        // Value starts at 30, make sure 30 reverts.
+        uint256 pbhExternalNullifier = PBHExternalNullifier.encode(30, month, year);
+        uint256 nullifierHash = 0;
+        vm.expectRevert();
```
Comment (0xKitsune) [inline contracts/test/PBHVerifierVerify.sol]: Can we update to expect the specific error?

File: contracts/test/PBHVerifierVerify.sol [hunk 3]
```diff
@@ -75,4 +69,37 @@ contract PBHVerifierVerify is PBHVerifierTest {
...
+        PBHVerifierImpl(address(pbhVerifier)).verifyPbhProof(
+            root, sender, nonce, testCallData, pbhExternalNullifier, nullifierHash, proof
+        );
+
+        // Increase numPbhPerMonth from non owner, expect revert
+        vm.prank(address(123));
+        vm.expectRevert();
```
Comment (0xKitsune) [inline contracts/test/PBHVerifierVerify.sol]: Same here, can we update to expect the specific error?

File: contracts/src/helpers/PBHExternalNullifier.sol (hunk 1 above)
Comment (0xForerunner) [inline contracts/src/helpers/PBHExternalNullifier.sol:60]: because the (0..=29) is equal to 30 entries. the pbhNonces are 0 indexed.

**Pull request merged**